hs/app/reach/Main.hs

```diff
@@ -2450,6 +2451,202 @@ whoami = command "whoami" $ info f fullDesc
   where
     f = pure . script $ write [N.text| echo "$whoami'" |]
 
+-- https://stackoverflow.com/a/60790430
+newtype GitHubGistResponse = GitHubGistResponse String
+instance FromJSON GitHubGistResponse where
+  parseJSON = withObject "GitHubGistResponse" $ \o -> GitHubGistResponse <$> o .: "html_url"
+
+support :: Subcommand
+support = command "support" $ info (pure step1) d
+  where
+    d = progDesc "Upload index.rsh and index.mjs to help us troubleshoot!"
+    splitByAmpersands :: BSLC8.ByteString -> [Text]
+    splitByAmpersands s = T.splitOn "&" (pack $ BSLC8.unpack s)
+    splitByEqualsSigns :: Text -> [Text]
+    splitByEqualsSigns s = T.splitOn (pack "=") s
+    process :: BSLC8.ByteString -> [[Text]]
+    process gitHubResponseString = map splitByEqualsSigns
+      $ splitByAmpersands gitHubResponseString
+    isDeviceCodePair :: [Text] -> Bool
+    isDeviceCodePair pair = head pair == pack "device_code"
+    isErrorPair :: [Text] -> Bool
+    isErrorPair pair = head pair == pack "error"
+    isUserCodePair :: [Text] -> Bool
+    isUserCodePair pair = head pair == pack "user_code"
+    isAccessTokenPair :: [Text] -> Bool
+    isAccessTokenPair pair = head pair == pack "access_token"
+
```

Comment:
!

Reply:
I've been "bear"-ish on deleting these because I thought separating related blocks would improve readability.

[b3f4bff](https://github.com/reach-sh/reach-lang/pull/762/commits/b3f4bffb806b4acffd004bbf6101c617667db4ab) addresses this.